src/util/wallet-utils/src/getAddress.ts

```diff
@@ -116,6 +128,35 @@ export const getAddressesAsigna: getAddresses = async (params) => {
   };
 };
 
+const getStxAddressAsigna: getAddresses = async () => {
+  const appConfig = new AppConfig(["store_write", "publish_data"]);
+  const userSession = new UserSession({ appConfig });
+  const result = await new Promise<FinishedAuthData>((res, rej) => {
+    authenticate(
+      {
+        userSession,
+        appDetails: {
+          name: "sBTC Bridge",
+          icon: window.location.origin + "/icon.png",
+        },
+        onFinish: res,
+        onCancel: rej,
+      },
+      AsignaIframeProvider as any,
+    );
+  });
+  const stxAddress = result.userSession.loadUserData().profile.stxAddress
+    .mainnet as string;
```

Comment:
What happens if the session data only has testnet address? I guess it will not happen (right now) with asigna, but maybe we should throw an error just for future-proofing

Reply:
the thing is and don't tell anybody asigna sets mainnet and testnet address to the same address, if it's testnet then it sets both of them to testnet if it's mainnet it sets both of them to mainnet musigs aren't portable through networks

Reply:
added a comment explaining this as well